protocol-units/da/m1/util/src/config/mod.rs

```diff
@@ -175,6 +175,15 @@ impl Config {
 			),
 		}
 	}
+
+	pub fn try_block_building_parameters(&self) -> Result<(u32, u64), anyhow::Error> {
```

Comment:
The return type should be a struct with descriptive field names.
What's the purpose of making this function fallible?